src/idiomatic/foundations-api-design/meaningful-doc-comments.md

```diff
@@ -0,0 +1,23 @@
+---
+minutes: 5
+---
+
+# Meaningful Doc Comments
+
+```rust,compile_fail
+/// API for Domain // ❌
+pub mod domain {}
+
+/// Function from A to B // ❌
+fn a_to_b(a: A) -> B {...}
+ 
+/// Does X // ❌
+fn do_x() {}
+```
+
+Doc comments are the most common source of documentation most developers will
+engage with.
+
+It's important to write doc comments that developers will appreciate reading,
+that gives them the information they are looking for and doesn't just re-state
+the obvious.
```

Comment:
Let's add some framing for the instructor (in the speaker notes).

- We are not here to provide conventional basic tech writing advice. There are many other resources for that.

- We are talking to an experienced audience, who already wrote doc comments in another programming language.

- We want to focus on higher level topics: what to include, and critically, what not to include in the documentation, as well as Rust specifics.